ledger/src/blockstore_db.rs

```diff
@@ -291,65 +291,33 @@ struct Rocks(rocksdb::DB, ActualAccessType, OldestSlot);
 
 impl Rocks {
     fn open(path: &Path, options: BlockstoreOptions) -> Result<Rocks> {
-        use columns::*;
-        let access_type = options.access_type;
-        let recovery_mode = options.recovery_mode;
+        let access_type = &options.access_type;
+        let recovery_mode = options.recovery_mode.clone();
 
         fs::create_dir_all(&path)?;
 
         // Use default database options
-        if matches!(access_type, AccessType::PrimaryOnlyForMaintenance) {
+        if disable_auto_compactions(access_type) {
             warn!("Disabling rocksdb's auto compaction for maintenance bulk ledger update...");
         }
-        let mut db_options = get_db_options(&access_type);
+        let mut db_options = get_db_options(access_type);
         if let Some(recovery_mode) = recovery_mode {
             db_options.set_wal_recovery_mode(recovery_mode.into());
         }
 
         let oldest_slot = OldestSlot::default();
-
-        // Get column family descriptors and names
-        let (cf_descriptor_shred_data, cf_descriptor_shred_code) =
-            new_cf_descriptor_pair_shreds::<ShredData, ShredCode>(
-                &options.shred_storage_type,
-                &access_type,
-                &oldest_slot,
-            );
-        let cfs = vec![
-            new_cf_descriptor::<SlotMeta>(&access_type, &oldest_slot),
-            new_cf_descriptor::<DeadSlots>(&access_type, &oldest_slot),
-            new_cf_descriptor::<DuplicateSlots>(&access_type, &oldest_slot),
-            new_cf_descriptor::<ErasureMeta>(&access_type, &oldest_slot),
-            new_cf_descriptor::<Orphans>(&access_type, &oldest_slot),
-            new_cf_descriptor::<BankHash>(&access_type, &oldest_slot),
-            new_cf_descriptor::<Root>(&access_type, &oldest_slot),
-            new_cf_descriptor::<Index>(&access_type, &oldest_slot),
-            cf_descriptor_shred_data,
-            cf_descriptor_shred_code,
-            new_cf_descriptor::<TransactionStatus>(&access_type, &oldest_slot),
-            new_cf_descriptor::<AddressSignatures>(&access_type, &oldest_slot),
-            new_cf_descriptor::<TransactionMemos>(&access_type, &oldest_slot),
-            new_cf_descriptor::<TransactionStatusIndex>(&access_type, &oldest_slot),
-            new_cf_descriptor::<Rewards>(&access_type, &oldest_slot),
-            new_cf_descriptor::<Blocktime>(&access_type, &oldest_slot),
-            new_cf_descriptor::<PerfSamples>(&access_type, &oldest_slot),
-            new_cf_descriptor::<BlockHeight>(&access_type, &oldest_slot),
-            new_cf_descriptor::<ProgramCosts>(&access_type, &oldest_slot),
-        ];
+        let cf_descriptors = Self::cf_descriptors(&options, &oldest_slot);
         let cf_names = Self::columns();
-        // The names and descriptors don't have to be in the same
-        // order, but there should be the same number of each.
-        assert_eq!(cfs.len(), cf_names.len());
```

Comment:
Moved to unit test

Reply:
Thanks for adding a unit test.  I think we can still keep the assertion, although RocksDB::open() will panic if we have a mismatch.

Or we could do it within a line as something like this so that it's no-op in the release code:

```
  assert_eq!(cfs.len(), Self::columns().len());
```

Reply:
The unit test does the exact same assertion given that `cfs = Rocks::cf_descriptors()` and `cf_names = Rocks::columns()`:
```
        assert_eq!(
            Rocks::columns().len(),
            Rocks::cf_descriptors(&options, &oldest_slot).len()
        );
```
I think keeping the same assertion in `Rocks::open()` would be redundant. The check exists to make sure if someone adds (or hypothetically, deletes) a new column family, they remember to update both functions. The unit test will fail if someone only updates one, so I think that adequately serves the purposes of notifying the developer that they missed something